core/src/main/java/org/apache/iceberg/view/ViewRepresentationParser.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+class ViewRepresentationParser {
+  static final String TYPE = "type";
+
+  private ViewRepresentationParser() {}
+
+  static void toJson(ViewRepresentation representation, JsonGenerator generator)
+      throws IOException {
+    Preconditions.checkArgument(representation != null, "Invalid view representation: null");
+    switch (representation.type()) {
+      case ViewRepresentation.Type.SQL:
+        SQLViewRepresentationParser.toJson((SQLViewRepresentation) representation, generator);
+        break;
+
+      default:
+        throw new IllegalArgumentException(
+            String.format(
+                "Cannot serialize unknown view representation type %s", representation.type()));
+    }
+  }
+
+  static String toJson(ViewRepresentation entry) {
+    return JsonUtil.generate(gen -> toJson(entry, gen), false);
+  }
+
+  static ViewRepresentation fromJson(String json, List<Schema> schemas) {
+    return JsonUtil.parseWithArg(json, ViewRepresentationParser::fromJson, schemas);
+  }
```

Comment:
What would happen in this model is that ViewMetadataParser would parse the list of schemas first and then pass that here. But trying to think of ways to simplify this. We may just want to go back to only having schema IDs for the representation API. Or even just serialize the entire schema alongside the representation. That should simplify a lot.